### DISCUSSION TITLE
Multiplatform support

### DISCUSSION THREAD

**Discussion by sdeleuze** [bendgk/effekt#3] [Ideas]
Since the library looks pretty small and seems currently JVM based, I am wondering if it would make sense to turn it to a multiplatform library with Kotlin/JS and potentially Kotlin/Wasm support (see my related blog post https://seb.deleuze.fr/the-huge-potential-of-kotlin-wasm/) to allow using it for web frontend development.

Any thoughts?

**Comment by bendgk** [comment]
Yes I'm down to try this!

I don't have much experience with with kotlin multiplatform, however I could try to study it over the next few days and Ill see what I can do.

Feel free to also give it a try and submit a PR!

**Reply by sdeleuze** [reply]
Nice thanks. Partially related questions, is this project implementing something similar to [Angular Signals](https://twitter.com/sarah_edo/status/1628065696247857152) (see [related demo](https://angular-signals.netlify.app/)) ?

**Reply by sdeleuze** [reply]
See https://twitter.com/youyuxi/status/1628214809631293440

**Reply by bendgk** [reply]
Signals could be another primitive I could add, But it seems like it should also be really easy for someone to build their own signal functionality using the existing primitives.

I've not worked much with Angular in the past, However, I am familiar with signals from solid.js

something similar to `useState()` from react or `createSignal()` from [solid.js](https://www.solidjs.com/docs/latest/api) could be added to look like so:

```kotlin
val (count, setCount) = createSignal(0)
```

will create a counter signal.

edit: although, In all honestly I need to consider whether signals fit into the design philosophy to begin with.  It just seems like another pattern for accomplishing the same reactivity that can be observed with `ref`, `computed`, and `watchEffect`

**Reply by bendgk** [reply]
Ive created another discussion so that we can talk about signals there, see #10

Lets keep this discussion related to multi-platform support 👍
